LTO can cause compilation to happen at the linking step, so we need to pass
CFLAGS to the linker as well. particularly, not having -fPIC in the link
command causes some tesseract C++ objects to fail to link.

reported upstream as https://bugs.ghostscript.com/show_bug.cgi?id=708940

diff --git a/configure.ac b/configure.ac
index 7935cde6d..300864964 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3174,11 +3174,11 @@ fi
 case $host in
     *-linux*|*-gnu)
       DYNAMIC_CFLAGS="-fPIC $DYNAMIC_CFLAGS"
-      GS_DYNAMIC_LDFLAGS="-shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(GS_SONAME_MAJOR)"
-      PCL_DYNAMIC_LDFLAGS="-shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(PCL_SONAME_MAJOR)"
-      XPS_DYNAMIC_LDFLAGS="-shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(XPS_SONAME_MAJOR)"
-      PDL_DYNAMIC_LDFLAGS="-shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(GPDL_SONAME_MAJOR)"
-      PDF_DYNAMIC_LDFLAGS="-shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(PDF_SONAME_MAJOR)"
+      GS_DYNAMIC_LDFLAGS="$DYNAMIC_CFLAGS -shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(GS_SONAME_MAJOR)"
+      PCL_DYNAMIC_LDFLAGS="$DYNAMIC_CFLAGS -shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(PCL_SONAME_MAJOR)"
+      XPS_DYNAMIC_LDFLAGS="$DYNAMIC_CFLAGS -shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(XPS_SONAME_MAJOR)"
+      PDL_DYNAMIC_LDFLAGS="$DYNAMIC_CFLAGS -shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(GPDL_SONAME_MAJOR)"
+      PDF_DYNAMIC_LDFLAGS="$DYNAMIC_CFLAGS -shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(PDF_SONAME_MAJOR)"
       if test x"$GCC" = x"yes"; then
         # GCC high level flag
         DYNAMIC_LIBS="-rdynamic"
